lib/core/storage.js

```diff
@@ -106,3 +107,67 @@ var store = module.exports.store = function(file, callback) {
     });
   });
 };
+
+/**
+ * @param  {String}     uri         S3 address or other
+ * @return {Boolean}                True URI matches s3://my-bucket/my-object-key
+ */
+var isS3Uri = module.exports.isS3Uri = function(uri) {
+  return _.startsWith(uri, 's3://');
+};
+
+/**
+ * Get object from Amazon S3
+ *
+ * @param  {String}     s3Uri                       URI of object in S3 (e.g., s3://my-bucket/my-object-key)
+ * @param  {Function}   callback                    Standard callback function
+ * @param  {Object}     callback.data               Data returned from Amazon S3
+ * @return {Object}                                 Callback return
+ */
+var getObject = module.exports.getObject = function(s3Uri, callback) {
+  var params = {
+    'Bucket': extractS3Bucket(s3Uri),
+    'Key': extractS3ObjectKey(s3Uri)
+  };
+
+  return callback(s3.getObject(params));
+};
+
+/**
+ * Get object metadata from Amazon S3
+ *
+ * @param  {String}     s3Uri                       URI of object in S3 (e.g., s3://my-bucket/my-object-key)
+ * @param  {Function}   callback                    Standard callback function
+ * @param  {Object}     callback.err                An error object, if any
+ * @param  {Object}     callback.data               Metadata returned from Amazon S3
+ */
+var getObjectMetadata = module.exports.getObjectMetadata = function(s3Uri, callback) {
+  var params = {
+    'Bucket': extractS3Bucket(s3Uri),
+    'Key': extractS3ObjectKey(s3Uri)
+  };
+
+  s3.headObject(params, function(err, metadata) {
+    if (err) {
+      return callback(err);
+    }
+
+    return callback(null, metadata);
+  });
+};
+
+/**
+ * @param  {String}     s3Uri          URI of object in S3 (e.g., s3://my-bucket/my-object-key)
+ * @return {String}                    Strictly the S3 bucket name
+ */
+var extractS3Bucket = function(s3Uri) {
+  return isS3Uri(s3Uri) ? s3Uri.split('/')[2] : null;
+};
+
+/**
+ * @param  {String}     s3Uri          URI of object in S3 (e.g., s3://my-bucket/my-object-key)
+ * @return {String}                    Strictly the S3 Object Key
+ */
+var extractS3ObjectKey = function(s3Uri) {
+  return isS3Uri(s3Uri) ? s3Uri.split('/').slice(3).join('/') : null;
+};
```

Comment:
@paulkerschen The amendment made in linked PR will be made above, too. Stand by.